### PR TITLE
Use native module on iOS to check if we've asked for push permissions

### PR DESCRIPTION
**Pull request by buoyad**

### CODE REVIEW

File: shared/actions/platform-specific.js.flow [hunk 1]
```diff
@@ -1,7 +1,6 @@
 // @flow
 declare export function checkPermissions(): Promise<boolean>
-declare export function setShownPushPrompt(): Promise<void>
-declare export function getShownPushPrompt(): Promise<string>
+declare export function getShownPushPrompt(): Promise<0 | 1>
```
Comment (akalin-keybase) [inline shared/actions/platform-specific.js.flow]: `Promise<boolean>` doesn't work?

Comment (buoyad) [reply shared/actions/platform-specific.js.flow]: It works, but a `1 | 0` is what's literally returned from the obj-C

**Comment by buoyad** [comment]
@akalin working on testing it on device now!

**Comment by buoyad** [comment]
@akalin-keybase Just tested it on device by enabling notifications, `Reset all settings`-ing, and allowing the app to request notifications again. It brought the native prompt up rather than going to settings.

File: shared/react-native/ios/Keybase/PushPrompt.m [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+
+#pragma mark - RCTBridgeModule
+
+RCT_EXPORT_MODULE(PushPrompt);
+
++ (BOOL)requiresMainQueueSetup
+{
+  return NO;
+}
+
```
Comment (chrisnojima) [inline shared/react-native/ios/Keybase/PushPrompt.m]: this is fine but just a general FYI, if we do native stuff you always have to make sure if it needs the main thread or not and if so to add the right methodQueue. In this case its safe to call this from any thread but many calls are main thread only still

**Pull request merged**